docs/content/guides/dagster-pipes/kubernetes.mdx

```diff
@@ -0,0 +1,243 @@
+
+- Created an asset named `k8s_pipes_asset`
+
+- Provided <PyObject object="AssetExecutionContext" /> as the `context` argument to the asset. This object provides access to system APIs such as resources, config, and logging.
+
+- Specified a resource for the asset to use, <PyObject module="dagster_k8s" object="PipesK8sClient" />, which is a pre-built Dagster resource that allows you to quickly get Pipes working with your Kubernetes workspace.
+
+  We also specified the following for the resource:
+
+  - `context` - The asset's `context` (<PyObject object="AssetExecutionContext" />) data
+  - `image` - The Kubernetes image we want to use
+  - `kubeconfig_file` - A kube-config file containing authentication information to pass to `kubernetes.config.load_kube_config`
+  - `kube_context` - A context argument to pass to `kubernetes.config.load_kube_config`
+
+- Passed the `context`, `image`, `kubeconfig_file`, and `kube_context` to the `run` method of <PyObject module="dagster_k8s" object="PipesK8sClient" />. This method submits the provided cluster information to the Kubernetes API and then runs the specified `image`.
+
+- Returned a <PyObject object="MaterializeResult" /> object representing the result of execution. This is obtained by calling `get_materialize_result` on the <PyObject module="dagster_pipes" object="PipesClientCompletedInvocation" /> object returned by `run` after the execution in Kubernetes has completed. **Note**: Execution can take several minutes even for trivial scripts due to cluster provisioning times.
+
+### Step 1.2: Create Dagster Definitions
+
+Next, you’ll add the asset and Kubernetes resource to your project’s code location via the <PyObject object="Definitions" /> object. This makes the resource available to other Dagster definitions in the project.
+
+Copy and paste the following to the bottom of `dagster_k8s_pipes.py`:
+
+```python
+# dagster_k8s_pipes.py
+
+defs = Definitions(
+  assets=[k8s_pipes_asset],
+  resources={
+    "k8s_pipes_client": PipesK8sClient(),
+  },
+)
+```
+
+At this point, `dagster_k8s_pipes.py` should look like the following:
+
+```python
+# dagster_k8s_pipes.py
+
+from dagster import AssetExecutionContext, Definitions, asset
+from dagster_k8s import PipesK8sClient
+
+
+@asset
+def k8s_pipes_asset(context: AssetExecutionContext, k8s_pipes_client: PipesK8sClient):
+  return k8s_pipes_client.run(
+      context=context,
+      image="pipes-materialize:latest",
+      kubeconfig_file=os.path.expanduser("~/.kube/config"),
+      kube_context="<YOUR_CONTEXT_NAME>"
+  ).get_materialize_result()
+
+
+defs = Definitions(
+  assets=[k8s_pipes_asset],
+  resources={
+    "k8s_pipes_client": PipesK8sClient(),
+  },
+)
+```
+
+---
+
+## Step 2: Write a script for execution on Kubernetes
+
+The next step is to write the code that will be executed on Kubernetes.
+
+TODO
+
+```python
+# my_python_script.py
+
+from dagster_pipes import open_dagster_pipes
+
+with open_dagster_pipes() as pipes:
```

Comment:
The context loader + message writer don't need to be specified since we're using defaults, correct?

Reply:
yea defaults should work